File: javascript/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var differenceHash = {};

    for (var i = 0; i < nums.length; i++) {
        var n = nums[i];
        var firstNumberIndex = differenceHash[n];

        if (firstNumberIndex === undefined) {
            differenceHash[target - n] = i;
        } else {
            return [firstNumberIndex, i];
        }
    }
};

File: ruby/two_sum.rb
def two_sum(nums, target)
    nums_hash = {}
    nums.each_with_index do |num, i|
        first_num_index = nums_hash[target - num]

        return [first_num_index, i] if first_num_index

        nums_hash[num] = i
    end
end
